tests/unit/test_solvers/test_casadi_algebraic_solver.py

```diff
@@ -97,10 +116,96 @@ def test_solve_with_input(self):
 
         # Solve
         solver = pybamm.CasadiAlgebraicSolver()
-        solution = solver.solve(model, np.linspace(0, 1, 10), inputs={"value": 7})
+        solution = solver.solve(model, np.linspace(0, 1, 10), inputs={"param": 7})
         np.testing.assert_array_equal(solution.y, -7)
 
 
+class TestCasadiAlgebraicSolverSensitivity(unittest.TestCase):
+    def test_solve_with_symbolic_input(self):
+        # Simple system: a single algebraic equation
+        var = pybamm.Variable("var")
+        model = pybamm.BaseModel()
+        model.algebraic = {var: var + pybamm.InputParameter("param")}
+        model.initial_conditions = {var: 2}
+        model.variables = {"var": var}
+
+        # create discretisation
+        disc = pybamm.Discretisation()
+        disc.process_model(model)
+
+        # Solve
+        solver = pybamm.CasadiAlgebraicSolver()
+        solution = solver.solve(model, [0], inputs={"param": "[sym]"})
```

Comment:
I'm getting a bit confused about what is "symbolic" :)  I could consider a `pybamm.InputParameter` to be symbolic, so I don't see why I would need to specify again that the input parameter is symbolic. I think the default behaviour of the inputs should be symbolic. That is, if I don't provide a value when `solve` is called, then this input parameter is treated as symbolic

Reply:
Ok - the solver has to know what shape the inputs have though, so I the model could keep track of this? Then the order in which the inputs were passed when solving will have to be read from the model to. Unless we always force `ProcessedSymbolicVariable.value()` to take a dictionary as input, and then the values get ordered in the same order as the model's inputs before being stacked?